docs/design/proposals/direct-blk-device-assignment.md

```diff
@@ -0,0 +1,264 @@
+ - resize <host-blk-device-name> <size> — update the mounted volume to <size> for the mounted filesystem backed by the given block device, <host-blk-device-name>.
+
+```
+The mountinfo object is defined as follows:
+```Golang
+type MountInfo struct {
+    // Device: source device path for this volume
+    Device string json:"device,omitempty"
+    // FsType: filesystem that needs to be used to mount the storage inside the VM
+    FsType string json:"fsType,omitempty"
+    // metadata field for easy extension
+    Metadata map[string]string json:"metadata,omitempty"
+    // Options: additional options that might be needed to mount the storage filesystem
+    Options []string json:"options,omitempty"
+}
+```
+The volume stas is defined as below following the [csi spec](https://github.com/container-storage-interface/spec/blob/master/lib/go/csi/csi.pb.go).
+```Golang
+type NodeGetVolumeStats struct {
+	// This field is OPTIONAL.
+	Usage []*VolumeUsage `json:"usage,omitempty"`
+	// Information about the current condition of the volume.
+	// This field is OPTIONAL.
+	// This field MUST be specified if the VOLUME_CONDITION node
+	// capability is supported.
+	VolumeCondition      *VolumeCondition `json:"volume_condition,omitempty"`
+}
+```
+
+#### Quick notes on implementation
+
+Instead of the CSI node driver writing the mount info into a `csiPlugin.json` file under the volume root, 
+as described in the original proposal. Here we propose that the CSI node driver passes the mount information to 
+the kata-runtime through a new kata-runtime command called `add-blk-mount-info`. The kata-runtime then write the mount 
+information to a `“`mount-info.json`”` file in a hardcoded location (e.g. `/run/kata-containers/shared/direct-block-devices/[block_dev_name]/`).
+
+When a Kata Pod sandbox is started, the containerd-kata-shim-v2 checks whether there is any mount identified 
+by its source path in the container specification that has a mount info file in the `direct-block-devices` directory. 
+If it exists, the containerd-kata-shim-v2 parses the mount info file and propagates it to the kata agent to carry out the mount. 
+Lastly,the containerd-kata-shim-v2 creates a file named by the sandbox id under the `direct-block-devices/[block_dev_name]/` 
+directory. The reason for adding a sandbox id file is to establish a mapping between the volume and the sandbox. 
+Later, when the kata-runtime handles the `get-stats` and `resize` commands, it can go to the volume’s directory, 
+find the sandboxes that mount this volume, and then identify the endpoint of the corresponding containerd-shim-kata-v2.
+
+### containerd-shim-kata-v2 changes
+containerd-shim-kata-v2 provides an API for sandbox management through a unix domain socket. Two new handlers are proposed:
+```
+New handlers:
+/fs-stats
+/fs-resize
+
+Example:
+
+curl —unix-socket <shim socket path> -I \
```

Comment:
Please don't fold lines - let GitHub's markdown renderer handle that.

Also, please put a `$ ` shell prompt at the start of every command and wrap those commands inside a ` ```bash ` block.